File: app/src/main/java/ke/choxxy/wordsearch/features/mainmenu/MainMenuActivity.java
package ke.choxxy.wordsearch.features.mainmenu;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModelProviders;
import android.content.Intent;
import android.os.Bundle;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import ke.choxxy.wordsearch.R;
import ke.choxxy.wordsearch.data.entity.MenuItem;
import ke.choxxy.wordsearch.easyadapter.MenuItemAdapter;
import ke.choxxy.wordsearch.features.ViewModelFactory;
import ke.choxxy.wordsearch.WordSearchApp;
import ke.choxxy.wordsearch.features.FullscreenActivity;
import ke.choxxy.wordsearch.features.gameplay.GamePlayActivity;
import ke.choxxy.wordsearch.features.settings.SettingsActivity;

import java.util.ArrayList;
import java.util.List;

import javax.inject.Inject;

import butterknife.BindArray;
import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class MainMenuActivity extends FullscreenActivity {

    @BindView(R.id.list) RecyclerView mRv;

    @BindArray(R.array.game_round_dimension_values)
    int[] mGameRoundDimVals;

    @Inject
    ViewModelFactory mViewModelFactory;
    MainMenuViewModel mViewModel;
    MenuItemAdapter mAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main_menu);

        ButterKnife.bind(this);
        ((WordSearchApp) getApplication()).getAppComponent().inject(this);

        mViewModel = ViewModelProviders.of(this, mViewModelFactory).get(MainMenuViewModel.class);
        // mViewModel.getOnGameThemeLoaded().observe(this, this::showGameThemeList);

        mAdapter = new MenuItemAdapter(new MenuItemAdapter.OnClickListener() {
            @Override
            public void onClick(@NonNull MenuItem menuItem) {
                newGame();
            }
        });

        mRv.setAdapter(mAdapter);
        mRv.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
        mAdapter.submitList(dummyList());
        mViewModel.loadData();
    }

    public List<MenuItem> dummyList(){

        String [] colors = new String[]{
                "#17DEEE",
                "#FF4162",
                "#F2E50B",
                "#21B20C",
                "#FF7F50",
                "#fdc675",
                "#d3fda1",
                "#96e3a5",
                "#41bec2"
        };

        String [] games = getResources().getStringArray(R.array.game_modes);

        List<MenuItem> dummy = new ArrayList<>();
        for(int i= 0; i < games.length;i++){
            dummy.add(new MenuItem(i, games[i], colors[i]));
        }
        return dummy;
    }

    @OnClick(R.id.settings_button)
    public void onSettingsClick() {
        Intent intent = new Intent(this, SettingsActivity.class);
        startActivity(intent);
    }


    public void newGame() {
        int dim = mGameRoundDimVals[(int) (Math.random()*mGameRoundDimVals.length)];
        Intent intent = new Intent(MainMenuActivity.this, GamePlayActivity.class);
        intent.putExtra(GamePlayActivity.EXTRA_ROW_COUNT, dim);
        intent.putExtra(GamePlayActivity.EXTRA_COL_COUNT, dim);
        startActivity(intent);
    }


}

File: app/src/main/java/co/africanwolf/hiddenwords/mainmenu/MainMenuActivity.kt
package co.africanwolf.hiddenwords.mainmenu

import android.content.Intent
import android.os.Bundle
import androidx.activity.viewModels
import androidx.recyclerview.widget.LinearLayoutManager
import co.africanwolf.hiddenwords.FullscreenActivity
import co.africanwolf.hiddenwords.data.entity.Game
import co.africanwolf.hiddenwords.data.entity.GameType
import co.africanwolf.hiddenwords.easyadapter.MenuItemAdapter
import co.africanwolf.hiddenwords.gameplay.GamePlayActivity
import co.africanwolf.hiddenwords.settings.SettingsActivity
import co.africanwolf.hiddenwords.R
import co.africanwolf.hiddenwords.databinding.ActivityMainMenuBinding
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.serialization.json.Json

@AndroidEntryPoint
class MainMenuActivity : FullscreenActivity() {

    private val mViewModel: MainMenuViewModel by viewModels()
    private lateinit var mAdapter: MenuItemAdapter
    private lateinit var binding: ActivityMainMenuBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainMenuBinding.inflate(layoutInflater)
        setContentView(binding.root)

        mAdapter = MenuItemAdapter(object : MenuItemAdapter.OnClickListener {
            override fun onClick(game: Game) {
                newGame(game)
            }
        })
        binding.list.adapter = mAdapter
        binding.list.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        mAdapter.submitList(menuItems())
        mViewModel.loadData()

        binding.settingsButton.setOnClickListener {
            val intent = Intent(this, SettingsActivity::class.java)
            startActivity(intent)
        }
    }

    private fun menuItems(): List<Game> {
        val colors = resources.getStringArray(R.array.menu_colors)
        val games = GameType.values()
        val gameList: MutableList<Game> = ArrayList()
        for (i in games.indices) {
            gameList.add(Game(i, games[i], colors[i], 100))
        }
        return gameList
    }

    fun newGame(game: Game) {

        val mGameRoundDimVals = resources.getIntArray(R.array.game_round_dimension_values)
        val dim = mGameRoundDimVals[(Math.random() * mGameRoundDimVals.size).toInt()]
        val intent = Intent(this@MainMenuActivity, GamePlayActivity::class.java)
        intent.putExtra(GamePlayActivity.EXTRA_ROW_COUNT, dim)
        intent.putExtra(GamePlayActivity.EXTRA_COL_COUNT, dim)
        intent.putExtra(
            GamePlayActivity.EXTRA_GAME_DATA,
            Json.encodeToString(Game.serializer(), game)
        )
        startActivity(intent)
    }
}
